benchmark/httpapi.py

```diff
@@ -32,15 +37,23 @@ class ResultNotFound(Exception):
     """
 
 
+class BadRequest(Exception):
+    """
+    Bad request parameters or content.
+    """
+
+
 @implementer(IBackend)
 class InMemoryBackend(object):
     """
-    The backend that simply drops all results.
-
-    :ivar dict results: Stored results by their identifiers.
+    The backend that keeps the results in the memory.
     """
     def __init__(self):
-        self._results = {}
+        def get_timestamp(result):
+            return timestamp_parser.parse(result['timestamp'])
+
+        self._results = OrderedDict()
```

Comment:
No, it does not have to be `OrderedDict` now that `SortedList` is used for keeping the order.
Fixed, thanks!